scripts/pre_push_hook.py

```diff
@@ -509,6 +518,13 @@ def main(args=None):
                         'Push failed, please correct the linting issues above.')
                     sys.exit(1)
 
+            mypy_check_status = start_mypy_checks()
```

Comment:
Is it possible to run mypy checks only for the changed files? (similar to lint checks)

Reply:
This is important as the time for type check is expected to increase rapidly, right? (And we are trying to reduce the pre-push load as much as possible)

Reply:
@DubeySandeep The time to run type checks won't be much as you can see in the PR description. This time further reduces as MyPy creates cache. 
In linters, we can run tests on all the files. In MyPy tests we have a list of files and directories which do not have type annotations so we would have to check every changed file whether it exists in the list or not. 
Do you think it is worth it? Thanks!

Reply:
And there was one more important point, a change somewhere in a file can result in mypy errors in other files. So I would say we should type check all the type annotated files where they are changed or unchanged.

Reply:
@DubeySandeep Let's suppose there are 2 files X and Y. X has function f definition and Y calls the function f.
Now in one PR, type annotation to file Y is added.
Now in a different PR, type annotations to file X is added. This can lead to mypy errors in file Y.
Now there are two cases:
1. Running pre-push type checks on all files: The error in file Y will be thrown and developer will fix them before pushing.
2. Running pre-push check on changed files: The error in file Y will not be thrown if it is not changed in the PR.

So I would highly suggest running pre-push type checks on all the type annotated files.

Reply:
@hardikkat24 Please edit the above msg and also explain with the example you discussed with me - how was mypy handling that error before (in PR for Y file) and what error would be shown had we been running mypy check locally for all files. (An example would help in understanding your argument better)

Reply:
@sajalasati Added the example.

Reply:
@hardikkat24 I'm not sure why pre-push check should rely on different PRs?

Reply:
As per our offline discussion, we're going to run checks for the entire codebase. Reasons:

1. Mypy cache for files are operated with file hash so the caches are going to be reliable and useful.
2. If a commit changes function definition then other modules calling such functions with invalid params won't get caught!